tests/integration/test_metrics.py

```diff
@@ -240,70 +390,85 @@ def test_session_metrics(mini_sentry, relay_with_processing, metrics_consumer):
     }
 
 
-def test_transaction_metrics(mini_sentry, relay_with_processing, metrics_consumer):
+@pytest.mark.parametrize(
+    "extract_metrics", [True, False], ids=["extract", "don't extract"]
+)
+@pytest.mark.parametrize(
+    "metrics_extracted", [True, False], ids=["extracted", "not extracted"]
+)
+def test_transaction_metrics(
+    mini_sentry,
+    relay_with_processing,
+    metrics_consumer,
+    metrics_extracted,
+    extract_metrics,
+):
     metrics_consumer = metrics_consumer()
 
-    for feature_enabled in (True, False):
-        relay = relay_with_processing(options=TEST_CONFIG)
-        project_id = 42
-        mini_sentry.add_full_project_config(project_id)
-        timestamp = datetime.now(tz=timezone.utc)
-
-        mini_sentry.project_configs[project_id]["config"]["features"] = (
-            ["organizations:metrics-extraction"] if feature_enabled else []
-        )
-
-        transaction = generate_transaction_item()
-        transaction["timestamp"] = timestamp.isoformat()
-        transaction["measurements"] = {
-            "foo": {"value": 1.2},
-            "bar": {"value": 1.3},
-        }
-        transaction["breakdowns"] = {"breakdown1": {"baz": {"value": 1.4},}}
+    relay = relay_with_processing(options=TEST_CONFIG)
+    project_id = 42
+    mini_sentry.add_full_project_config(project_id)
+    timestamp = datetime.now(tz=timezone.utc)
 
-        relay.send_event(42, transaction)
+    mini_sentry.project_configs[project_id]["config"]["features"] = (
+        ["organizations:metrics-extraction"] if extract_metrics else []
+    )
 
-        # Send another transaction:
-        transaction["measurements"] = {
-            "foo": {"value": 2.2},
-        }
-        transaction["breakdowns"] = {"breakdown1": {"baz": {"value": 2.4},}}
-        relay.send_event(42, transaction)
+    transaction = generate_transaction_item()
+    transaction["timestamp"] = timestamp.isoformat()
+    transaction["measurements"] = {
+        "foo": {"value": 1.2},
+        "bar": {"value": 1.3},
+    }
+    transaction["breakdowns"] = {"breakdown1": {"baz": {"value": 1.4},}}
 
-        if not feature_enabled:
-            message = metrics_consumer.poll(timeout=None)
-            assert message is None, message.value()
+    #: The `metrics_extracted` header is ignored for transactions for now.
```

Comment:
Why? Is there going to be another PR? In INGEST-250 we say we want it to work for transactions

Reply:
There is a separate story for that: https://getsentry.atlassian.net/browse/INGEST-332.